ios/engine/KMEI/KeymanEngine/Classes/KeymanWebDelegate.swift

```diff
@@ -16,7 +16,7 @@ protocol KeymanWebDelegate: class {
   /// - Parameters:
   ///   - numCharsToDelete: The number of UTF-16 code units to delete before inserting the new text.
   ///   - newText: The string to insert.
-  func insertText(_ keymanWeb: KeymanWebViewController, numCharsToDelete: Int, newText: String)
+  func insertText(_ keymanWeb: KeymanWebViewController, numCharsToLeftDelete: Int, newText: String, numCharsToRightDelete: Int)
```

Comment:
I'd have kinda liked this to have a different parameter order:

```suggestion
  func insertText(_ keymanWeb: KeymanWebViewController, numCharsToLeftDelete: Int, numCharsToRightDelete: Int, newText: String)
```

Reply:
From Web:

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/web/source/kmwbase.ts#L633-L638

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/web/source/dom/domOverrides.ts#L36-L38

From Android:

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java#L2351-L2353

That said, both of these arose during 14.0 - there's no evidence of them in 13.0.  In their original versions, they closely matched the iOS function above:  `deleteLeft`, then `newText`.

Admittedly, the order was chosen b/c it's a new parameter, and new things - especially potentially-undefined parameters (b/c JS/TS) - go on the right-hand side.  That said, there _is_ a beauty to the order:

abc _de_ | _fg_ hij

With respect to the caret, we first delete-left, then insert the text at the caret's position, then delete-right if needed after the caret.  Though, I suppose that temporal order doesn't particularly matter - it's largely transitive as long as delete-lefts come before text insertion.  It does make sense with spatial order, though.

That said... I don't mind changing it... but only if we also change the parameter order in those locations.  And I think that's best left to a separate PR.

Reply:
Okay with that refactor being a separate PR

Reply:
Tracked as https://github.com/keymanapp/keyman/issues/4529.